main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -0,0 +1,244 @@
+              stopWithDuplicateUrl.csvRowNumber(),
+              stopWithDuplicateUrl.stopId(),
+              stopWithDuplicateUrl.stopUrl(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).routeId(),
+              routesByUrlMap.get(stopWithDuplicateUrl.stopUrl()).csvRowNumber()
+          ));
+    });
+  }
+
+  /**
+   * Maps {@code GtfsAgency}s by there URLs if provided.
+   *
+   * @param stopTable the {@code GtfsStopTableContainer} to extract {@code GtfsStop} from
+   * @return routes from {@code GtfsStopTableContainer}s mapped by there {@code routes.route_url}
+   * (in lower case) if provided.
+   */
+  private Map<String, GtfsStop> stopsByUrlMap(GtfsStopTableContainer stopTable) {
+    Map<String, GtfsStop> stopsByUrl = new HashMap<>();
+    stopTable.getEntities().forEach(stop -> {
+      if (stop.hasStopUrl()) {
+        if (stopsByUrl.get(stop.stopUrl()) == null) {
+          stopsByUrl.put(stop.stopUrl().toLowerCase(), stop);
+        }
+      }
+    });
+    return stopsByUrl;
+  }
+
+  /**
+   * Maps {@code GtfsRoute}s by their URLs if provided.
+   *
+   * @param routeTable the {@code GtfsRouteTableContainer} to extract {@code GtfsRoute} from
+   * @return routes from {@code GtfsRouteTableContainer}s mapped by there {@code routes.route_url}
+   * (in lower case) if provided.
+   */
+  private Map<String, GtfsRoute> routesByUrlMap(GtfsRouteTableContainer routeTable) {
+    Map<String, GtfsRoute> routesByUrl = new HashMap<>();
+    routeTable.getEntities().forEach(route -> {
+      if (route.hasRouteUrl()) {
+        if (routesByUrl.get(route.routeUrl()) == null) {
+          routesByUrl.put(route.routeUrl().toLowerCase(), route);
+        }
+      }
+    });
+    return routesByUrl;
+  }
+
+  /**
+   * Maps {@code GtfsAgency}s by their URLs if provided.
+   *
+   * @param agencyTable the {@code GtfsAgencyTableContainer} to extract {@code GtfsAgency} from
+   * @return agencies from {@code GtfsAgencyTableContainer}s mapped by there URLs (in lower case) if
+   * provided.
+   */
+  private Map<String, GtfsAgency> agenciesByUrlMap(GtfsAgencyTableContainer agencyTable) {
+    Map<String, GtfsAgency> agenciesByUrl = new HashMap<>();
+    agencyTable.getEntities().forEach(agency -> {
+      if (agency.hasAgencyUrl()) {
+        if (agenciesByUrl.get(agency.agencyUrl()) == null) {
+          agenciesByUrl.put(agency.agencyUrl().toLowerCase(), agency);
+        }
+      }
+    });
+    return agenciesByUrl;
+  }
+
+  /**
+   * A {@code GtfsStop} has the same value for {@code stops.stop_url} as a record from
+   * "routes.txt".
+   * <p>
+   * {@code SeverityLevel.WARNING}
+   */
+  static class SameStopAndRouteUrlNotice extends ValidationNotice {
+
+    @SchemaExport
+    SameStopAndRouteUrlNotice(long csvRowNumber, String stopId, String stopUrl,
+        String routeId, long routeCsvRowNumber) {
+      super(
+          ImmutableMap.of(
+              "csvRowNumber", csvRowNumber,
```

Comment:
Here and elsewhere: it is unclear which table `csvRowNumber` is referring to. Please rename it to `stopCsvRowNumber` or `routeCsvRowNumber`.